lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts

```diff
@@ -0,0 +1,54 @@
+/*!
+ * @license
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BaseApi } from './base.api';
+import { throwIfNotDefined } from '../../../assert';
+import { ContentPagingQuery, NodeChildAssociationPaging } from '../../content-rest-api';
+
+/**
+ * Legal Holds service.
+ *
+ * @module LegalHoldApi
+ */
+export class LegalHoldApi extends BaseApi {
+    /**
+     * List of legal holds
+     *
+     * @param filePlanId The identifier of a file plan. You can also use the -filePlan- alias.
+     * @param opt Optional parameters
+     * @returns Promise<NodeChildAssociationPaging>
+     */
+    getHolds(filePlanId: string = '-filePlan-', opts?: ContentPagingQuery): Promise<NodeChildAssociationPaging> {
+        throwIfNotDefined(filePlanId, 'filePlanId');
+
+        const pathParams = {
+            filePlanId
+        };
+
+        const queryParams = {
+            skipCount: opts?.skipCount,
+            maxItems: opts?.maxItems
+        };
+
+        return this.get({
+            path: '/file-plans/{filePlanId}/holds',
+            pathParams,
+            queryParams,
+            returnType: NodeChildAssociationPaging
```

Comment:
Why this API should return `NodeChildAssociationPaging`? I think it returns `HoldPaging` which looks totally different

Reply:
The `/file-plans/{filePlanId}/holds` endpoint has nothing to do with `NodeChildAssociationPaging` so it's not correct class to be used in this case, please check what should be returned by it in the API definition

Reply:
created `HoldPaging` class